### DISCUSSION TITLE
signal: cannot open links in firefox with --private (snap)

### DISCUSSION THREAD

**Discussion by compassconslt** [netblue30/firejail#6082] [Q&A] [answered]
I am running Signal Desktop in Firejail via the "--private" option to have multiple profiles. Signal will ask for the user to verify the account sometimes when starting a new chat with a new contact. To verify, a button is presented to the user, which when not run in Firejail, opens the default browser to a webpage to verify that the user is human. 

In Firejail, running the command:

`firejail /usr/bin/signal-desktop`

..results in a successful button press i.e. the button opens Firefox as prescribed without any modifications.

In Firejail "--private" the user experience is that the button is presented, but nothing happens when pressed. Here is the error message as observed in the log:

```
`panic: permission denied

goroutine 1 [running]:
github.com/snapcore/snapd/snapdtool.ExecInSnapdOrCoreSnap()
	/build/snapd-KuK39h/snapd-2.58+22.04.1/snapdtool/tool_linux.go:205 +0x40e
main.main()
	/build/snapd-KuK39h/snapd-2.58+22.04.1/cmd/snap/main.go:443 +0x45`
```

It looks like adding the "--private" flag forces Firejail to use Firefox. I tried changing the default browser to Chrome as well as adding the noblacklist and whitelist options for google-chrome, to no avail.

I believe the error is related to snap permissions. This error happens regardless if I setup for google-chrome or Firefox. I have both listed in the "noblacklist" of signal-desktop.profile. 

Update:  This was a snap issue -- Ubuntu 22.04 comes with Firefox installed by default via snap. To remedy the situation no changes were needed for Firejail. Instead uninstalling the snap version of Firefox and installing the APT version solved this problem.

**Comment by kmk3** [comment, marked as answer]
@compassconslt on [Nov 10, 2023](https://github.com/netblue30/firejail/discussions/6082#discussion-5835099):

> Update: This was a snap issue -- Ubuntu 22.04 comes with Firefox installed by
> default via snap. To remedy the situation no changes were needed for
> Firejail. Instead uninstalling the snap version of Firefox and installing the
> APT version solved this problem.
